SDWebImage/Core/SDDiskCache.m

```diff
@@ -240,18 +242,41 @@ - (nullable NSString *)cachePathForKey:(NSString *)key {
 
 - (NSUInteger)totalSize {
     NSUInteger size = 0;
-    NSDirectoryEnumerator *fileEnumerator = [self.fileManager enumeratorAtPath:self.diskCachePath];
-    for (NSString *fileName in fileEnumerator) {
-        NSString *filePath = [self.diskCachePath stringByAppendingPathComponent:fileName];
-        NSDictionary<NSString *, id> *attrs = [self.fileManager attributesOfItemAtPath:filePath error:nil];
-        size += [attrs fileSize];
+
+    // Use URL-based enumerator instead of Path(NSString *)-based enumerator to reduce
+    // those objects(ex. NSPathStore2/_NSCFString/NSConcreteData) created during traversal.
+    // Even worse, those objects are added into AutoreleasePool, in background threads, 
+    // the time to release those objects is undifined(according to the usage of CPU)
+    // It will truely consumes a lot of VM, up to cause OOMs. 
+    NSDirectoryEnumerationOptions options = (NSDirectoryEnumerationOptions)0;
+    if (@available (iOS 13, *)) {
+        // NSDirectoryEnumerationProducesRelativePathURLs causes the NSDirectoryEnumerator 
+        // to always produce file path URLs relative to the directoryURL. This can reduce 
+        // the size of each URL object returned during enumeration.
+        options |= NSDirectoryEnumerationProducesRelativePathURLs;
+    }
+    
+    @autoreleasepool {
+        NSURL *pathURL = [NSURL fileURLWithPath:self.diskCachePath isDirectory:YES];
```

Comment:
Why use `isDirectory: YES` here ? Without reading the attributes from file system